### DISCUSSION TITLE
504 error while fetching ref tags from Rest API

### DISCUSSION THREAD

**Discussion by snoir** [community/community#87691] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I'm having 504 errors while doing a GET on this particular URI : https://api.github.com/repos/aws/aws-sdk-go-v2/git/refs/tags

This repository has more than 30,000 tags, I've tried with success on other repositories, with way less tags

**Comment by Naimulnobel** [comment]
The 504 error you're encountering is a Gateway Timeout error. This typically happens when a server that is acting as a gateway or proxy did not receive a timely response from another server that it was accessing while attempting to load the web page or fill another request by the browser.

In your case, the GitHub API server is likely having trouble retrieving all the tags for the `aws/aws-sdk-go-v2` repository due to the large number of tags. 

Here are a few potential solutions:

1. **Pagination**: The GitHub API supports pagination. You can use the `page` and `per_page` query parameters to retrieve a subset of the tags at a time. For example, to get the first 100 tags, you could use `https://api.github.com/repos/aws/aws-sdk-go-v2/git/refs/tags?page=1&per_page=100`.

2. **Conditional Requests**: If you're repeatedly fetching the same data, you can use conditional requests to avoid unnecessary requests. The GitHub API will return a `304 Not Modified` status if the data hasn't changed since the last request.

3. **Rate Limiting**: Ensure that you're not hitting the rate limit for the GitHub API. Authenticated requests get a higher rate limit than unauthenticated requests.

If none of these solutions work, it might be a limitation of the GitHub API with large repositories. In this case, you might want to contact GitHub Support for further assistance.
